pkg/apis/core/validation/validation.go

```diff
@@ -5195,17 +5196,31 @@ func ValidatePodTemplateSpec(spec *core.PodTemplateSpec, fldPath *field.Path, op
 	return allErrs
 }
 
-func ValidateReadOnlyPersistentDisks(volumes []core.Volume, fldPath *field.Path) field.ErrorList {
+// ValidateReadOnlyPersistentDisks validates pd reads and writes
+func ValidateReadOnlyPersistentDisks(featureGate featuregate.FeatureGate, volumes, oldVolumes []core.Volume, fldPath *field.Path, isNew bool) field.ErrorList {
 	allErrs := field.ErrorList{}
+
+	if !featureGate.Enabled(features.SkipReadOnlyValidationGCE) {
+		return field.ErrorList{}
+	}
+
+	isWriteablePD := func(vol *core.Volume) bool {
+		return vol.GCEPersistentDisk != nil && !vol.GCEPersistentDisk.ReadOnly
+	}
+
+	if !isNew {
+		for i := range oldVolumes {
+			if isWriteablePD(&oldVolumes[i]) {
```

Comment:
Please do add comments - I included them in my example because this logic is not exactly obvious WHY :)

Reply:
this whether the validation is for a new use of read-write or an existing use.

Reply:
I removed it. I may think it's unnecessary verification

Reply:
*I* know why we have this block here.  But eventually I will forget why, and someone new to the code will not understand why.  That's why I commented in my example above:

```
+	// If this was already being violated, don't flag it (i.e. this is a
+	// rollback from a version which allows it by default)
+	for i := range oldVolumes {
+		if isWritablePD(&oldVolumes[i]) {
+			// The new volumes are allowed to use read-write.
```